### DISCUSSION TITLE
Open Server Panel 6.2.1

### DISCUSSION THREAD

**Discussion by OSPanel** [OSPanel/OpenServerPanel#25] [Анонсы]
### Что нового

- **GUI**: Графический интерфейс настроек программы со встроенной справкой и удобный просмотрщик логов.
- **Языки**: Вьетнамский, Испанский, Итальянский, Немецкий.
- **Новые модули**: Mailpit, MariaDB-11.4, MariaDB-11.7, MongoDB-8.0, MySQL-8.4, PHP-8.4, PostgreSQL-17, RabbitMQ-4.0, Redis-7.4, Smtp4dev.
- **Проверка сторонних конфигурационных файлов**: Теперь программа проверяет наличие файлов `my.cnf` и `my.ini` на диске C:, что помогает избежать конфликтов со штатно установленным ПО.
- **Возможность указания HTTP и HTTPS портов для проектов**: Теперь вы можете гибко настраивать порты для веб-проектов, обеспечивая оптимальную конфигурацию и безопасность.
- **Опция `short_project_title`**: При включении этой опции в консоли будет отображаться только имя проекта вместо детального заголовка окна (вкладки).
- **Опция `use_system_environment`**: Возможность принудительно включать системную среду в окружение всех проектов.
- **Опция `additional_ssl_hosts`** (для модулей): Опция позволяет указать IP-адреса и домены, которые будут добавлены в автоматически генерируемый ssl-сертификат.
- **Опция `ignore_shutdown_all`** (для модулей): При включении этой опции модуль не будет выключаться по команде выключения всех модулей.
- **Селектор проектов**: При вводе команды `osp project` без указания имени проекта вы сможете выбрать нужный проект из общего списка с помощью ввода его номера в этом списке. При вводе команды `osp project auto`, если вы уже находитесь в каталоге проекта, то проект будет активирован автоматически. Если текущий каталог назначен нескольким проектам, то будет активирован первый проект по списку.
- **Темы оформления иконок меню**: Добавлены дополнительные варианты стилизации иконок в меню для улучшения визуального оформления интерфейса.
- **Настройка размера шрифта меню**: Пользователи с нарушениями зрения имеют возможность настроить индивидуальный размер шрифта в меню интерфейса, если использование системного масштабирования Windows по тем или иным причинам (например, из-за конфликтов с другими приложениями или личных предпочтений) для них неудобно или неприменимо.
- **Поддержка системной отрисовки меню**: Опция включает системную отрисовку меню для совместимости с программами скрин-ридерами, обеспечивая доступность для пользователей с нарушениями зрения.
- **Архив проектов**: Добавлен специальный раздел для неактивных проектов (`enabled = off`). В меню можно управлять статусом проектов: деактивировать (поместить в архив) или активировать (восстановить из архива).
- **Категории проектов**: Реализована система категоризации проектов в меню программы. Категория указывается в конфигурационном файле проекта. Доступна специальная категория `hidden` для служебных ресурсов - проекты в этой категории не отображаются в основном меню. Это удобно для размещения административных инструментов, таких как phpMyAdmin.

### Изменения

- **Автоматическое присвоение IP-адресов**: Для модулей, использующих локальные IP-адреса формата 127.127.126.XXX, теперь используется автоматическая настройка. Это упрощает создание и распространение новых модулей для пользователей, заинтересованных в их интеграции. При необходимости IP-адрес можно вручную указать в параметрах модуля.
- **Возвращение php-расширения Ioncube**: Поддержка Ioncube снова доступна, однако будьте осторожны с изменением дефолтных настроек, так как это может вызвать сбои при запуске дочерних процессов PHP через планировщик или вручную.
- **Отдельный блок настроек прокси для Composer**: Настройки прокси для Composer теперь вынесены в отдельный блок, что упрощает их конфигурацию и управление.
- **Выделение части PHP-дополнений**: Blackfire, FFMpeg, Ghostscript и ImageMagick, теперь вынесены в отдельные дополнения. Это решение позволяет оптимизировать использование дискового пространства и устанавливать только необходимые компоненты.
- **Оптимизация HTTP-заголовков в Apache и Nginx**: Популярные HTTP-заголовки по умолчанию отключены и собраны в единый блок, что позволит избежать дублирования заголовков и сбоев в работе некоторых CMS.
- **Самоуправление подпроцессами PHP-CGI**: FCGI-версии модулей PHP теперь управляют своими подпроцессами самостоятельно, без использования `php-cgi-spawner.exe`, что улучшает производительность и стабильность.
- **Зависимость FCGI-версий модулей PHP от обычных модулей PHP**: FCGI-версии модулей PHP теперь используют файлы обычных версий PHP, что позволило значительно сэкономить занимаемое ими место на диске. Для использования в шаблонах конфигурации FCGI-версий модулей PHP добавлена шаблонная переменная `{parent_module}`.
- **Подсветка значков проектов в меню**: Значки проектов теперь подсвечиваются в зависимости от статуса, аналогично модулям, что улучшает визуальное восприятие и удобство навигации.
- **Упрощение работы с меню**: Имена текущих профилей для модулей и имена назначенных модулей для проектов теперь выводятся непосредственно в меню модуля или проекта.
- **Увеличение размера стека в Apache и PHP 8**: Размер стека по умолчанию увеличен до 8 МБ, что позволяет обрабатывать более сложные запросы и сценарии.
- **Оптимизация параметра ThreadsPerChild**: Значение ThreadsPerChild снижено до 32, что уменьшает потребление памяти после увеличения ThreadStackSize.
- **Увеличение xdebug.max_nesting_level в PHP 7.x**: Параметр `xdebug.max_nesting_level` увеличен до 512, что позволяет выполнять более глубокие вложенные вызовы без ошибок.
- **Удаление встроенного SMTP-сервера**: Встроенный SMTP-сервер был исключён из состава программы в связи с интеграцией модулей Mailpit и Smtp4dev, которые обеспечивают аналогичную функциональность, а также предлагают расширенные возможности и улучшенную гибкость при работе с электронной почтой.
- **Изменение в логике обработки доменных псевдонимов**: Псевдонимы, заданные в виде IP-адресов, более не рассматриваются как IP-адреса доменов, находящихся под управлением веб-сервера.
- **Обработка доменных псевдонимов по новому**: Доменные псевдонимы (алиасы) без символа точки в имени больше не дополняются именем хоста проекта и обрабатываются как полноценные доменные имена.
- **Важные операции SysPrepTool**: Все важные операции, выполняемые SysPrepTool, теперь скрыты и не могут быть отключены.

### Исправления

- **Проблемы с запуском Shell для модулей MariaDB и MongoDB**: Исправлен запуск Shell для корректной работы с этими модулями.
- **Удаление устаревших модулей MySQL-5.5 и MongoDB 3.0-3.2**: Устаревшие версии модулей удалены из-за нестабильной работы, что повышает общую надёжность системы.
- **Корректировка конфигурации mod_watch и SNMP**: Доработаны конфигурации для mod_watch (Apache) и SNMP (PHP), чтобы избежать появления служебных каталогов этих расширений на системном диске.
- **Устранение ошибки "Request-URI Too Long"**: Скорректированы настройки Apache и Nginx для предотвращения этой ошибки.
- **Исправление команды запуска проекта**: Команда "`osp project <DOMAIN> start`" теперь работает корректно.
- **Коррекция цвета трей-иконки**: Исправлен оттенок красного цвета трей-иконки программы для более чёткого восприятия.
- **Исправление работы планировщика заданий**: Планировщик заданий теперь правильно интерпретирует стандартные настройки плана.
- **Поддержка масштабирования интерфейса**: Исправлено отображение меню программы на системах с включённым масштабированием интерфейса, таких как 4k мониторы.
- **Исправления запуска программы из корня диска**: Запуск программы теперь возможен непосредственно из корня диска.
- **Улучшение работы консольных команд**: Исправлена работа некоторых консольных команд и взаимодействие со средами в консоли.
- **Коррекция работы IDN-конвертера**: Исправлен IDN-конвертер, обеспечив корректное преобразование доменов, содержащих дефис.

**Comment by aivuser** [comment]
Здравствуйте. Сейчас у меня версия 6.3.5. Не работает функция "Категории проектов". В конфиг проекта добавляю 
category = "Рабочие" (пробовал кириллицу и латиницу, с кавычками и без)
Где посмотреть пример? Вдруг я что-то упускаю или есть некие ограничения...

**Reply by OSPanel** [reply]
> Здравствуйте. Сейчас у меня версия 6.3.5. Не работает функция "Категории проектов". В конфиг проекта добавляю category = "Рабочие" (пробовал кириллицу и латиницу, с кавычками и без) Где посмотреть пример? Вдруг я что-то упускаю или есть некие ограничения...

Категория используется только если в настройках меню включена группировка проектов.

**Reply by aivuser** [reply]
Да, спасибо, разобрался методом научного тыка, но было бы неплохо такое документировать хотя бы в примечании к опции category. Она при этом работает как-то странновато и неочевидно. Например, она группирует проекты, деля их по "частям" доменов, если category не задана. Так, если есть несколько папок вида <что-то_там>.local, то появится группа "local". Это было неожиданно, но не проблема.

П.С. 
Ещё обнаружил, что если проект статичный: HTML/CSS/JS, то без включения любого сервера (Apache(модуль PHP)/Nginx) он работать не будет, хотя в списке проектов появляется и его можно открыть в браузере из меню. В целом не проблема, просто заметка.
